src/turbomind/models/llama/LlamaBatch.cc

```diff
@@ -111,12 +111,17 @@ void LlamaBatch::DisableInvalidRequests(Requests& infer_reqs, Requests& kill_req
         }
     };
 
-    auto validate = [&occur](auto& reqs, const char* type) {
+    auto validate = [&](auto& reqs, const char* type) {
         for (const auto& r : reqs) {
             if (occur[r->id] > 1) {
                 TM_LOG_ERROR("Skip conflicting %s request for ID %lu", type, r->id);
                 r->ec = Request::kConflict;
             }
+            if (param_.enable_prefix_caching && !r->session.start_flag) {
+                // Prefix caching is incompatible with interactive mode
+                TM_LOG_ERROR("Skip inconsistent %s request for ID %lu", type, r->id);
+                r->ec = Request::kInconsistency;
+            }
```

Comment:
I think we could check `session.step == 0` here, so that we don't need to modify line 233-236, 262 and maybe 1018.